pkg/agent/run.go

```diff
@@ -28,16 +28,20 @@ var AuthToken string
 // Period is the number of seconds between scans
 var Period uint
 
+// Agent will run only once if OneShot is enabled
+var OneShot bool
+
 // CredentialsPath is where the agent will try to loads the credentials. (Experimental)
 var CredentialsPath string
 
+
 // Run starts the agent process
 func Run(cmd *cobra.Command, args []string) {
 	ctx := context.Background()
-
-	for {
-		gatherAndPostData(ctx)
+	gatherAndPostData(ctx)
+	for OneShot != true {
 		time.Sleep(time.Duration(Period) * time.Second)
+		gatherAndPostData(ctx)
```

Comment:
This is fine as it is, however I prefer to not repeat the "meat" of the loop. It's only one function
call today, but we might want to change that later. Therefore I would choose to use a for loop
with `if OneShot { break }` in it. You don't have to change anything though.